product/infrastructure/fake/searchservice.go

```diff
@@ -23,29 +23,81 @@ func (s *SearchService) Inject(
 	return s
 }
 
-//Search returns Products based on given Filters
+// Search returns Products based on given Filters
 func (s *SearchService) Search(ctx context.Context, filters ...searchDomain.Filter) (*domain.SearchResult, error) {
-	documents := make([]searchDomain.Document, 0)
-	hits := make([]domain.BasicProduct, 0)
+	hits := s.findProducts(ctx, filters)
+	currentPage := s.findCurrentPage(filters)
+	facets, selectedFacets := s.createFacets(filters)
 
-	for _, marketPlaceCode := range s.productService.GetMarketPlaceCodes() {
-		product, _ := s.productService.Get(ctx, marketPlaceCode)
-		if product == nil {
-			continue
+	documents := make([]searchDomain.Document, len(hits))
+	for i, hit := range hits {
+		documents[i] = hit
+	}
+
+	return &domain.SearchResult{
+		Result: searchDomain.Result{
+			SearchMeta: searchDomain.SearchMeta{
+				Query:          "",
+				OriginalQuery:  "",
+				Page:           currentPage,
+				NumPages:       10,
+				NumResults:     len(hits),
+				SelectedFacets: selectedFacets,
+				SortOptions:    nil,
+			},
+			Hits:       documents,
+			Suggestion: []searchDomain.Suggestion{},
+			Facets:     facets,
+		},
+		Hits: hits,
+	}, nil
+}
+
+// SearchBy returns Products prefiltered by the given attribute (also based on additional given Filters)
+func (s *SearchService) SearchBy(ctx context.Context, _ string, _ []string, _ ...searchDomain.Filter) (*domain.SearchResult, error) {
+	return s.Search(ctx)
+}
+
+func (s *SearchService) findProducts(ctx context.Context, filters []searchDomain.Filter) []domain.BasicProduct {
+	products := make([]domain.BasicProduct, 0)
+
+	// - try finding product by marketPlaceCode given in query
+	if query, found := s.filterValue(filters, "q"); found {
+		var product domain.BasicProduct
+		if len(query) > 0 {
+			product, _ = s.productService.Get(ctx, query[0])
+		}
+		if product != nil {
+			products = append(products, product)
+		}
```

Comment:
Why not return here immediately? With this implementation, we currently can't test queries that don't match any product which could be a valid case you want to simulate with a fake service.